### ISSUE TITLE
Necessary columns missing a column that should be necessary

### ISSUE THREAD

**Issue by lgray** [dask-contrib/dask-awkward#169]
Found a bug that will be run into regularly, here's a repro (using dask-awkward@main):

```python3
import uproot
import dask_awkward as dak

x = uproot.dask({"https://github.com/CoffeaTeam/coffea/blob/master/tests/samples/nano_dy.root?raw=true": "Events"})
print(dak.necessary_columns(dak.broadcast_arrays(x.fixedGridRhoFastjetAll, x.Jet_pt)))
```
prints:
```
{'from-uproot-d77a4895f123e53606d66a8430b53410': ['Jet_pt']}
```
As opposed to needing the data from both columns!

@agoose77 @jpivarski since this could possibly include an issue with typetracer reports as well, but I can't find where it's failing.

FYI - I can fix it by manually touching all the data that goes into dak.broadcast_arrays.

**Comment by douglasdavis** [comment]
First small bite :)

**Comment by lgray** [comment]
Exactly! That's why I moved it here :-) Sorry for the title mishap doing too many things today.

**Comment by douglasdavis** [comment]
This _may_ be fixed by #173 - would love a second set of eyes on it

**Comment by lgray** [comment]
```
>>> import uproot
>>> import dask_awkward as dak
>>> x = uproot.dask({"https://github.com/CoffeaTeam/coffea/blob/master/tests/samples/nano_dy.root?raw=true": "Events"})
>>> print(dak.necessary_columns(dak.broadcast_arrays(x.fixedGridRhoFastjetAll, x.Jet_pt)))
{'from-uproot-ef7eb37d96679681ca5f47c612501a51': ['Jet_pt', 'fixedGridRhoFastjetAll']}
```

with #173 merged :-)
